### ISSUE TITLE
Blockwishlist - the wishlists have a public URL that is not protected enough

### ISSUE THREAD

**Issue by lmeyer1** [PrestaShop/PrestaShop#29815]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Wishlist pages are publicly accessible at an URL of the form `/module/blockwishlist/view?id_wishlist=<id>`, and they are indexable (no noindex tag).

### Expected behavior

The fact they are publicly available si questionable to me, but it can be a feature to let a customer share his list with others on social media.
But there are two things that should fixed:
- The page should have a `<meta name="robots" content="noindex">` to make sure these pages are never indexed.
- The URL should not contain a plain index, that allows someone to browse all existing wishlists, but a instead a token that cannot be guessed.

### Steps to reproduce

1. As a customer, add a product to a wishlist.
2. Navigate to _My account - My wishlists_
3. Copy the URL
4. Log out
5. Access the URL while not connected

### PrestaShop version(s) where the bug happened

1.7.8.7

### PHP version(s) where the bug happened

7.3

### If your bug is related to a module, specify its name and its version

blockwishlist 2.1.2

**Comment by florine2623** [comment]
Hello @lmeyer1 ,

On blockwishlist module, a wishlist can be shared only via the share link that looks like : `myshop.com/en/module/blockwishlist/view?token=38586C798D8BAFD2`

If I share a link through my wishlist in my customer acccunt, I cannot access via if I am not logged in as customer : `myshop.com/en/module/blockwishlist/view?id_wishlist=1`
<img width="1135" alt="Screenshot 2022-09-29 at 17 02 04" src="https://user-images.githubusercontent.com/16019289/193067717-23c75be9-bf9c-469a-acda-7fd238cf08b9.png">

Can you specify what more you want through this issue ? ^^
Waiting for your feedback!
Thanks!

**Comment by lmeyer1** [comment]
@florine2623
I expect the error you show. But I don't get this error, I get the display of the wishlist. The affected website will be publicly accessible on monday. I will you send a link to a wishlist.
I don't have the time to investigate why I do experience the problem why you don't. I have PS 1.7.8.7 and the latest version of blockwishlist 2.1.2.

**Comment by florine2623** [comment]
You are using PS 1.6.7.6 ? 

On my end, I tested with 1.7.8.7 as you wrote in the issue and blockwishlist v2.1.2.

Let me know when you have more info!

**Comment by lmeyer1** [comment]
@florine2623 Sorry that was a typo. I'm using 1.7.8.7.

**Comment by PrestaEdit** [comment]
I confirm the issue.

The link to share have a token but the link on the title will give you the ID of the list ; but there is a protection about access to the customer to show you a 403 error if it's not yours.

**Comment by hibatallahAouadni** [comment]
Hello @lmeyer1 

I didn't manage to reproduce the issue with PS1787, see the attached screen record below:

[issue_29815.webm](https://user-images.githubusercontent.com/23308427/195150238-8f2ae1cc-ad6a-4dc7-849a-dce29845cc29.webm)

Is your shop a fresh install or was it upgraded from an old version?
Waiting for your feedback.

Thanks!

**Comment by lmeyer1** [comment]
After more testing, I found out, that the URL `/module/blockwishlist/view?id_wishlist=<id>` is not accessible publicly, but that it remains only accessible on browser where the owner of the wishlist was logged before. 
The problem is not so much with the wishlist URL anymore, but with the fact, that after logging out, one still has access to some resources owned by the logged out user, that should no longer be accessible. 
I remember that I discussed the latter problem on slack.

I close this one. The bug reported does not exist. The observations were misinterpreted.
